content/de/client-server-architecture.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Client-Server-Architektur
+status: Feedback erwünscht
+category: Konzept
+---
+
+## Was es ist
+
+Bei einer Client-Server-Architektur ist eine Anwendung in zwei oder mehr Komponenten aufgeteilt. Dabei wird unterschieden zwischen Clients und Servern. 
+Ein Client fordert die Erledigung einer Aufgabe bei einem Server an und der Server wiederum nimmt diese Anfrage entgegen und bearbeitet die Aufgabe. 
+Ein Beispiel dafür ist Google Mail. Der Client ist in diesem Fall die Webanwendung bzw. die UI, die in dem Browser des Nutzers ausgeführt wird. 
+Mit einem Klick auf "Email senden" wird die geschriebene Email and den Server gesendet (Dienst zum Emails versenden), der in der Cloud von Google läuft und die Email an den Empfänger weiterleitet.
+
+Dies steht im Gegensatz zu eigenständigen Anwendungen (z. B. Desktop-Anwendungen), die die gesamte Arbeit an einem Ort erledigen. 
+So kann beispielsweise ein Textverarbeitungsprogramm wie Microsoft Word vollständig auf dem Computer des Nutzers installiert und ausgeführt werden.
+
+## Welches Problem es löst
+
+Eine Client-Server-Architektur löst eine große Herausforderung, die in sich geschlossene Anwendungen mit sich bringen: Regelmäßige Aktualisierungen. 
+Bei einer eigenständigen Anwendung müssten die Benutzer bei jeder Aktualisierung die neueste Version herunterladen und installieren. 
+Zum Beispiel müssten sich die Benutzer regelmäßig den kompletten Produktkatalog von Amazon auf ihre Computer herunterladen. 
+Außerdem vereinfacht es die gemeinsame Nutzung von Daten. So müssen Daten nicht mehr verschickt oder sogar auf physischen Datenträgern weitergegeben werden.
```

Comment:
Weiß nicht recht ob ich das auf Anhieb in Verbindung zu einander bringen kann und ob es wirklich so ein gutes Beispiel ist. Vielleicht könnte man Skalierung noch einbringen?!

Reply:
Skalierung ist eine gute Idee.
Aber ich sehe nicht was an der gemeinsamen Nutzung von Daten falsch ist. Für mich gehört das schon hier hin, stammt auc hvon der englischen Originalversion.

Reply:
@iamNoah1 hab da irgendwie bisschen Probleme, das mit der Skalierung als Problem zu formulieren. Könnt ihr mir vlt einen Denkanstoß geben oder vlt doch weglassen? @bcubk

Reply:
Denkanstoß wäre der folgende Gedanke: Wenn wir eine Standalone Applikation haben, die auf der lokalen Maschine installiert werden muss und die Applikation immer größer wird, setzt das eine stärkere Maschine voraus. Wenn wir die Applikation aber in Client und Server trennen, können wir das Backend hochskalieren und auf der Maschine müsste dann nur noch der schmale client laufen.